---
title: "Salmon_QC on all datasets"
output: 
  html_document:
    keep-md: yes
date: "2023-09-05"
---

# Knitr Setup
```{r setup-chunk, include=FALSE}
knitr::opts_chunk$set(dev = "png",
                      dpi = 300,
                      echo = FALSE,
                      cache = TRUE)
```


# Load required libraries
```{r}
library(Rqc)
library(tximport)
library(EnsDb.Mmusculus.v79)
library(DESeq2)
library(RNAseqQC)
library(ggplot2)
library(apeglm)
library(pheatmap)
library(RColorBrewer)
library(AnnotationHub)
library(EnhancedVolcano)
library(biomaRt)
```

# Use Rqc to perform QC on FASTQ files from RO RNAseq
## Load FASTQ files into Rqc
```{r}
folder <- "/PATH/TO/FASTQs"
qa <- rqc(path = folder, pattern = "fastq.gz")
```

## Generate file information table
```{r}
Info <- perFileInformation(qa)
write.csv(Info, file = "/PATH/TO/perFileInformation.csv")
```

## Per read mean quality distribution
```{r}
BP <- rqcReadQualityBoxPlot(qa)
BP + theme_light()
```

## Average read quality
```{r}
rqcReadQualityPlot(qa)
```


#Load expression data - output from Salmon
```{r}
dir <- "/PATH/TO/ReprodEpith_SalmonQuant/"
samples <- read.table(file.path(dir,"Samples.txt"), header=TRUE)
samples
files <- file.path(dir,samples$Sample, "quant.sf")
names(files) <- samples$Condition
```

```{r}
file.exists(files)
head(files)
```

```{r}
txdb <- makeTxDbFromGFF(file = "/PATH/TO/Mmus.gtf.gz", format = "gtf")
```

```{r}
k <- keys(txdb, keytype = "TXNAME")
tx2gene <- select(txdb, k, "GENEID", "TXNAME")
txi <- tximport(files, type="salmon", tx2gene=tx2gene, ignoreTxVersion=TRUE)
```

# Perform differential gene expression analysis

```{r}
head(txi)
```


```{r}
ddsTxi <- DESeqDataSetFromTximport(txi,
                                   colData = samples,
                                   design = ~ Condition)
```

## RNAseq QC 
### Pre-filter transcripts with >= 50 reads
```{r}
keep <- rowSums(counts(ddsTxi)) >= 50
ddsTxi <- ddsTxi[keep,]
```

### Plot total counts per sample
```{r TotalCountsPlot-RO, dev = c("png", "tiff")}
plot_total_counts(ddsTxi)
```

### BoxPlot of normalized counts for each sample
```{r NormalizedCountsBoxPlot-RO, dev = c("png", "tiff"), fig.height = 12, fig.width = 15,}
ddsTxi <- estimateSizeFactors(ddsTxi)

par(cex.axis=01,mar=c(9,2,1,1))
boxplot(log10(counts(ddsTxi,normalized=TRUE)+1), las=2, col = c("#3fb0bf", "#3fb0bf", "#3fb0bf", "#d11382","#d11382","#d11382", "#2ECC71", "#2ECC71", "#2ECC71", "#F4D03F", "#F4D03F", "#F4D03F", "#D35400", "#D35400", "#D35400", "#8E44AD", "#8E44AD", "#8E44AD" ))

```

## Variance-stabilizing transformation
```{r}
vsd <- vst(ddsTxi)
class(vsd)
```

```{r}
assay(vsd)[1:3,1:6]
```

## Principal components plot
```{r PCA-Plot-RO, dev = c("png", "tiff"), fig.height = 12, fig.width = 15,}
pcaData <- plotPCA(vsd, intgroup = c( "Condition", "Sample"), returnData = TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(x = PC1, y = PC2, color = Condition)) +
geom_point(size =10) +
xlab(paste0("PC1: ", percentVar[1], "% variance")) +
ylab(paste0("PC2: ", percentVar[2], "% variance")) +
coord_fixed() +
scale_colour_manual(values = c("#D35400", "#F4D03F", "#8E44AD", "#d11382", "#2ECC71" , "#3fb0bf" )) + 
theme_light()+ theme(text=element_text(size=30))
```

## Heatmap of the sample-to-sample distances
```{r Heatmap-ALL, dev = c("png", "tiff"), fig.height = 12, fig.width = 15,}

sampleDists <- dist(t(assay(vsd)))
sampleDistMatrix <- as.matrix(sampleDists)
rownames(sampleDistMatrix) <- paste(vsd$Name, sep="-")
colnames(sampleDistMatrix) <- NULL
colors <- colorRampPalette( rev(brewer.pal(9, "YlGnBu")) )(255)
pheatmap(sampleDistMatrix,
         clustering_distance_rows=sampleDists,
         clustering_distance_cols=sampleDists,
         col=colors, cex = 1.2) 
```

## Run Differential Gene Expression Analysis - fetal vs. adult RO
```{r}
dds <- DESeq(ddsTxi)
resRO_Ovi <- results(dds, contrast=c("Condition","Adult_RO","Adult_Ovi"))
resRO_OSE <- results(dds, contrast=c("Condition","Adult_RO","Adult_OSE"))
resRO_Ova <- results(dds, contrast=c("Condition","Adult_RO","Adult_Ovary"))
resRO_fRO <- results(dds, contrast=c("Condition","Adult_RO","Fetal_RO"))
resfRO_fOva <- results(dds, contrast=c("Condition","Fetal_RO","Fetal_Ovary"))

```

```{r}
resRO_Ovi 
resRO_OSE 
resRO_Ova 
resRO_fRO 
resfRO_fOva 
```


### Sort results by the smallest p value
```{r}
resRO_OviOrdered <- resRO_Ovi [order(resRO_Ovi$pvalue),]
resRO_OSEOrdered <- resRO_OSE [order(resRO_OSE$pvalue),]
resRO_OvaOrdered <- resRO_Ova [order(resRO_Ova$pvalue),]
resRO_fROOrdered <- resRO_fRO [order(resRO_fRO$pvalue),]
resfRO_fOvaOrdered <- resfRO_fOva [order(resfRO_fOva$pvalue),]
```

### Summarize results
```{r}
summary(resRO_OviOrdered)
summary(resRO_OSEOrdered)
summary(resRO_OvaOrdered)
summary(resRO_fROOrdered)
summary(resfRO_fOvaOrdered)

```

### MA_Plot of results with apeglm LFC shrinkage
```{r}
resultsNames(dds)
```

```{r}
dds$Condition = relevel(dds$Condition, ref = "Adult_Ovary")
```

```{r}
dds <- nbinomWaldTest(dds)
resultsNames(dds)
```

```{r}
lfc.resRO_Ova <- lfcShrink(dds, coef="Condition_Adult_RO_vs_Adult_Ovary", type="apeglm")
```

```{r MA_Plot-RO, dev = c("png", "tiff"), fig.height = 5, fig.width = 15,}
plotMA(lfc.resRO_Ova, ylim=c(-5,5),colSig = "#d11382", main="Adult RO vs. Adult Ovary", cex=0.6, cex.lab = 1.2, font = 2 ) + theme_light()
```

```{r}
dds$Condition = relevel(dds$Condition, ref = "Adult_OSE")
```

```{r}
dds <- nbinomWaldTest(dds)
resultsNames(dds)
```

```{r}
lfc.resRO_OSE <- lfcShrink(dds, coef="Condition_Adult_RO_vs_Adult_OSE", type="apeglm")
lfc.resfRO_OSE <- lfcShrink(dds, coef="Condition_Fetal_RO_vs_Adult_OSE", type="apeglm")
```


```{r MA_Plot-RO, dev = c("png", "tiff"), fig.height = 5, fig.width = 15,}
plotMA(lfc.resRO_OSE, ylim=c(-5,5),colSig = "#d11382", main="Adult RO vs. Adult OSE", cex=0.6, cex.lab = 1.2, font = 2 ) + theme_light()
plotMA(lfc.resfRO_OSE, ylim=c(-5,5),colSig = "#3fb0bf", main="Fetal RO vs. Adult OSE", cex=0.6, cex.lab = 1.2, font = 2 ) + theme_light()
```

```{r}
dds$Condition = relevel(dds$Condition, ref = "Adult_Ovi")
```

```{r}
dds <- nbinomWaldTest(dds)
resultsNames(dds)
```

```{r}
lfc.resRO_Ovi <- lfcShrink(dds, coef="Condition_Adult_RO_vs_Adult_Ovi", type="apeglm")

```


```{r MA_Plot-RO, dev = c("png", "tiff"), fig.height = 5, fig.width = 15,}
plotMA(lfc.resRO_Ovi, ylim=c(-5,5),colSig = "#d11382", main="Adult RO vs. Adult Ovi", cex=0.6, cex.lab = 1.2, font = 2 ) + theme_light()

```

```{r}
dds$Condition = relevel(dds$Condition, ref = "Fetal_Ovary")
```

```{r}
dds <- nbinomWaldTest(dds)
resultsNames(dds)
```

```{r}

lfc.resfRO_fOva <- lfcShrink(dds, coef="Condition_Fetal_RO_vs_Fetal_Ovary", type="apeglm")
```


```{r MA_Plot-RO, dev = c("png", "tiff"), fig.height = 5, fig.width = 15,}

plotMA(lfc.resfRO_fOva, ylim=c(-5,5),colSig = "#3fb0bf", main="Fetal RO vs. Fetal Ovary", cex=0.6, cex.lab = 1.2, font = 2 ) + theme_light()
```


```{r}
dds$Condition = relevel(dds$Condition, ref = "Fetal_RO")
```

```{r}
dds <- nbinomWaldTest(dds)
resultsNames(dds)
```

```{r}

lfc.resRO_fRO <- lfcShrink(dds, coef="Condition_Adult_RO_vs_Fetal_RO", type="apeglm")
```


```{r MA_Plot-RO, dev = c("png", "tiff"), fig.height = 5, fig.width = 15,}

plotMA(lfc.resRO_fRO, ylim=c(-5,5),colSig = "#d11382", main="Adult RO vs. Fetal RO", cex=0.6, cex.lab = 1.2, font = 2 ) + theme_light()
```


# Prepare data for export
## Use AnnotationHub to add info to the results
```{r}
ah <- AnnotationHub()
```

### Query AnnotationHub to find Mus musculus entry
```{r}
query(ah, c("OrgDb","Mus musculus"))
```

### Pull the Mus musculus record
```{r}
Mm <- ah[["AH95960"]]
```

### Inspect the concordance with the dds results table
```{r}
table(rownames(resRO_fROOrdered) %in% keys(Mm, "ENSEMBL"))
```

## Add gene symbols to the results table and export as .csv
```{r}
resRO_OviOrdered$symbol <- mapIds(Mm, rownames(resRO_OviOrdered), column="SYMBOL", keytype="ENSEMBL")
head(resRO_OviOrdered)
write.csv(as.data.frame(resRO_OviOrdered),
	file="RO_Ovi_results.csv")

resRO_OSEOrdered$symbol <- mapIds(Mm, rownames(resRO_OSEOrdered), column="SYMBOL", keytype="ENSEMBL")
head(resRO_OSEOrdered)
write.csv(as.data.frame(resRO_OSEOrdered),
	file="RO_OSE_results.csv")

resRO_OvaOrdered$symbol <- mapIds(Mm, rownames(resRO_OvaOrdered), column="SYMBOL", keytype="ENSEMBL")
head(resRO_OvaOrdered)
write.csv(as.data.frame(resRO_OvaOrdered),
	file="RO_Ova_results.csv")

resfRO_fOvaOrdered$symbol <- mapIds(Mm, rownames(resfRO_fOvaOrdered), column="SYMBOL", keytype="ENSEMBL")
head(resfRO_fOvaOrdered)
write.csv(as.data.frame(resfRO_fOvaOrdered),
	file="fRO_fOva_results.csv")

resRO_fROOrdered$symbol <- mapIds(Mm, rownames(resRO_fROOrdered), column="SYMBOL", keytype="ENSEMBL")
head(resRO_fROOrdered)
write.csv(as.data.frame(resRO_fROOrdered),
	file="RO_fRO_results.csv")
```

# Volcano Plots
## Add gene symbols to the lfcShrink results tables 
```{r}
lfc.resfRO_fOva$symbol <- mapIds(Mm, rownames(lfc.resfRO_fOva), column="SYMBOL", keytype="ENSEMBL")

lfc.resRO_fRO$symbol <- mapIds(Mm, rownames(lfc.resRO_fRO), column="SYMBOL", keytype="ENSEMBL")

lfc.resRO_OSE$symbol <- mapIds(Mm, rownames(lfc.resRO_OSE), column="SYMBOL", keytype="ENSEMBL")

lfc.resRO_Ovi$symbol <- mapIds(Mm, rownames(lfc.resRO_Ovi), column="SYMBOL", keytype="ENSEMBL")

lfc.resRO_Ova$symbol <- mapIds(Mm, rownames(lfc.resRO_Ova), column="SYMBOL", keytype="ENSEMBL")
```

### EnhancedVolcano Plots of results with apeglm LFC shrinkage
```{r VolcanoPlot-RO, dev = c("png", "tiff"), fig.height = 8, fig.width = 5,}
deseq.volcano <- function(res, datasetName, topN = 30) {
  # Sort the results by p-value and select the top N genes
  res_sorted <- res[order(res$padj), ]
  top_genes <- res_sorted[1:topN, ]
  
  # Create a vector of labels for the top genes, and NA for others
  labels <- ifelse(row.names(res) %in% row.names(top_genes), as.character(res$symbol), NA)

  return(EnhancedVolcano(res,
                         x = 'log2FoldChange',
                         y = 'padj',
                         lab = labels,
                         
                         title = paste(datasetName, "Adult RO vs Fetal RO"),
                         subtitle = bquote(italic('FDR <= 0.05 and absolute FC >= 2')),
                         labSize = 4,
                         pointSize = 2,
                         axisLabSize = 12,
                         titleLabSize = 14,
                         subtitleLabSize = 10,
                         captionLabSize = 12,
                         
                         pCutoff = 10e-10,
                         FCcutoff = 2,
                         
                         # Customize the legend
                         legendLabSize = 9,  # Adjust the legend font size
                         legendPosition = "top"  # Adjust the legend position
                         ))
}

deseq.volcano(res = lfc.resRO_fRO, datasetName = "L2FC")
```


```{r VolcanoPlot-RO, dev = c("png", "tiff"), fig.height = 8, fig.width = 5,}

deseq.volcano <- function(res, datasetName, topN = 30) {
  # Sort the results by p-value and select the top N genes
  res_sorted <- res[order(res$padj), ]
  top_genes <- res_sorted[1:topN, ]
  
  # Create a vector of labels for the top genes, and NA for others
  labels <- ifelse(row.names(res) %in% row.names(top_genes), as.character(res$symbol), NA)

  return(EnhancedVolcano(res,
                         x = 'log2FoldChange',
                         y = 'padj',
                         lab = labels,
                         
                         title = paste(datasetName, "Adult RO vs Adult OSE"),
                         subtitle = bquote(italic('FDR <= 0.05 and absolute FC >= 2')),
                         labSize = 4,
                         pointSize = 2,
                         axisLabSize = 12,
                         titleLabSize = 14,
                         subtitleLabSize = 10,
                         captionLabSize = 12,
                         
                         pCutoff = 10e-10,
                         FCcutoff = 2,
                         
                         # Customize the legend
                         legendLabSize = 9,  # Adjust the legend font size
                         legendPosition = "top"  # Adjust the legend position
                         ))
}

deseq.volcano(res = lfc.resRO_OSE, datasetName = "L2FC")
```

```{r VolcanoPlot-RO, dev = c("png", "tiff"), fig.height = 8, fig.width = 5,}
deseq.volcano <- function(res, datasetName, topN = 30) {
  # Sort the results by p-value and select the top N genes
  res_sorted <- res[order(res$padj), ]
  top_genes <- res_sorted[1:topN, ]
  
  # Create a vector of labels for the top genes, and NA for others
  labels <- ifelse(row.names(res) %in% row.names(top_genes), as.character(res$symbol), NA)

  return(EnhancedVolcano(res,
                         x = 'log2FoldChange',
                         y = 'padj',
                         lab = labels,
                         
                         title = paste(datasetName, "Adult RO vs Adult Ovary"),
                         subtitle = bquote(italic('FDR <= 0.05 and absolute FC >= 2')),
                         labSize = 4,
                         pointSize = 2,
                         axisLabSize = 12,
                         titleLabSize = 14,
                         subtitleLabSize = 10,
                         captionLabSize = 12,
                         
                         pCutoff = 10e-10,
                         FCcutoff = 2,
                         
                         # Customize the legend
                         legendLabSize = 9,  # Adjust the legend font size
                         legendPosition = "top"  # Adjust the legend position
                         ))
}

deseq.volcano(res = lfc.resRO_Ova, datasetName = "L2FC")
```

```{r VolcanoPlot-RO, dev = c("png", "tiff"), fig.height = 8, fig.width = 5,}
deseq.volcano <- function(res, datasetName, topN = 30) {
  # Sort the results by p-value and select the top N genes
  res_sorted <- res[order(res$padj), ]
  top_genes <- res_sorted[1:topN, ]
  
  # Create a vector of labels for the top genes, and NA for others
  labels <- ifelse(row.names(res) %in% row.names(top_genes), as.character(res$symbol), NA)

  return(EnhancedVolcano(res,
                         x = 'log2FoldChange',
                         y = 'padj',
                         lab = labels,
                         
                         title = paste(datasetName, "Adult RO vs Adult Oviduct"),
                         subtitle = bquote(italic('FDR <= 0.05 and absolute FC >= 2')),
                         labSize = 4,
                         pointSize = 2,
                         axisLabSize = 12,
                         titleLabSize = 14,
                         subtitleLabSize = 10,
                         captionLabSize = 12,
                         
                         pCutoff = 10e-10,
                         FCcutoff = 2,
                         
                         # Customize the legend
                         legendLabSize = 9,  # Adjust the legend font size
                         legendPosition = "top"  # Adjust the legend position
                         ))
}
deseq.volcano(res = lfc.resRO_Ovi, datasetName = "L2FC")
```

```{r VolcanoPlot-RO, dev = c("png", "tiff"), fig.height = 8, fig.width = 5,}
deseq.volcano <- function(res, datasetName, topN = 30) {
  # Sort the results by p-value and select the top N genes
  res_sorted <- res[order(res$padj), ]
  top_genes <- res_sorted[1:topN, ]
  
  # Create a vector of labels for the top genes, and NA for others
  labels <- ifelse(row.names(res) %in% row.names(top_genes), as.character(res$symbol), NA)

  return(EnhancedVolcano(res,
                         x = 'log2FoldChange',
                         y = 'padj',
                         lab = labels,
                         
                         title = paste(datasetName, "Fetal RO vs Fetal Ovary"),
                         subtitle = bquote(italic('FDR <= 0.05 and absolute FC >= 2')),
                         labSize = 4,
                         pointSize = 2,
                         axisLabSize = 12,
                         titleLabSize = 14,
                         subtitleLabSize = 10,
                         captionLabSize = 12,
                         
                         pCutoff = 10e-10,
                         FCcutoff = 2,
                         
                         # Customize the legend
                         legendLabSize = 9,  # Adjust the legend font size
                         legendPosition = "top"  # Adjust the legend position
                         ))
}
deseq.volcano(res = lfc.resfRO_fOva, datasetName = "L2FC")
```
```{r}
txi_df <- as.data.frame(txi)
txi_df$symbol <- mapIds(Mm, rownames(txi_df), column="SYMBOL", keytype="ENSEMBL")
head(txi_df)
```

```{r}
write.csv(as.data.frame(txi_df),
	file="/PATH/TO/TPMs_ALL.csv")
```

 


### SessionInfo
```{r}
sessionInfo()
```

